src/components/feedback/Dialog.tsx

```diff
@@ -33,6 +47,15 @@ type ComponentProps = {
    * Restore focus to previously-focused element when unmounted/closed
    */
   restoreFocus?: boolean;
+
+  /**
+   * Providing this has the next implications:
+   * - The component will be used to wrap the Dialog contents.
+   * - If initialFocus === 'auto', the Dialog will be focused once the open
+   *   transition has finished.
+   * - onClose will be invoked after the close transition has finished.
+   */
```

Comment:
We'll want to tighten down this comment before we land this as most of this is implementation detail. The consumer probably doesn't need to care about most of this :)